### PR TITLE
fix: the signature unmatch in async trait function

### PR DESCRIPTION
**Pull request by BowenXiao1999**
## What's changed and what's your intention?

When we use `state_table.iter()` in a async trait function, it shows a confusing compiler error about signature mismatch. I think it's compiler's problem but I do not get too many evidence. I just removed the unnecessary closure mentioned in the error and it seems work.

@wcy-fdu can you please check that it fixs your problem? I have tested it on my version.

The compiler error looks like:
image

Because the `.map` looks unnecessary, so I just removed it.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Review by skyzh — APPROVED** [review]
```
let mut mem_table_iter = mem_table_iter
             .map(|(k, v)| Ok::<_, StorageError>((k, v)))
             .peekable();
```

Using this can help implement this with async_stream. As we are not using async_stream for now, I'd be okay with this change.

**Comment by BowenXiao1999** [comment]
I also provide a PR if you want to test locally #2999 : https://github.com/singularity-data/risingwave/pull/2999/commits/2a7e5156b1293e7ac6d26f449a3f6ece71bfbc27

File: src/storage/src/table/state_table.rs [hunk 1]
```diff
@@ -422,7 +418,7 @@ impl<S: StateStore> StateTableRowIter<S> {
                 (Some(_), Some(_)) => {
                     // Throw the error.
                     cell_based_table_iter.next().await.unwrap()?;
-                    mem_table_iter.next().unwrap()?;
+                    mem_table_iter.next().unwrap();
```
Comment (BugenZhao) [inline src/storage/src/table/state_table.rs:421]: We may remove this line since there's no error to throw for memtable iterator now. :)

**Review by BugenZhao — APPROVED** [review]

**Review by wcy-fdu — APPROVED** [review]
This PR also works in my case!
Good Job!

File: src/storage/src/table/state_table.rs (hunk 1 above)
Comment (BowenXiao1999) [reply src/storage/src/table/state_table.rs:421]: I think it's seperate problem so I submit another PR. #3071

Comment (BowenXiao1999) [reply src/storage/src/table/state_table.rs:421]: Emmm think for twice and I think you are right.. It should be done in this PR. anyway..

Comment (BowenXiao1999) [reply src/storage/src/table/state_table.rs:421]: Actually previous it also can not be Error. It's always map with Ok.

Comment (BugenZhao) [reply src/storage/src/table/state_table.rs:421]: Previously there’re some decoding stuff in the `map` body so it can be error. Seems this has changed.😁

**Pull request merged**